### DISCUSSION TITLE
Error on reconnect to BleDevice interface "GATT Server is disconnected".

### DISCUSSION THREAD

**Discussion by faydogan** [capacitor-community/bluetooth-le#416] [Q&A]
I am using Angular 13+, when I try to store BleDevice, I connected, in a service which is injected at root level, and try to reconnect through the stored object it throws an exception 

"_DOMException: GATT Server is disconnected. Cannot retrieve services. (Re)connect first with `device.gatt.connect`._".

```typescript
  let device = this._MyService.bleDevice;
  const WEIGHT_SERVICE = "0000181b-0000-1000-8000-00805f9b34fb";
  const WEIGHT_CHARACTERISTIC = "00002a9c-0000-1000-8000-00805f9b34fb";
  await BleClient.connect(device.deviceId, (deviceId) => onDisconnect(deviceId)).then(async () => {
     ...
  });
```
Is there any way to have a reconnect to a device at least for the same browser tab? or Can this be a feature to implement?

I read somethıng like this which Web Bluetooth API allows and tried to implement the logic with this package but still same issue. https://googlechrome.github.io/samples/web-bluetooth/automatic-reconnect.html

**Comment by pwespi** [comment]
Hello. Thanks for using this plugin.

I'm not sure I fully understand what's going on in your case. But if in the `onDisconnect` callback you call `BleClient.connect` again, that would be equivalent to the code you linked.
